src/Build/Evaluation/Expander.cs

```diff
@@ -3476,10 +3478,27 @@ private bool TryExecuteWellKnownFunction(out object returnVal, object objectInst
                     }
                 }
 
+                if (Traits.Instance.LogPropertyFunctionsRequiringReflection)
+                {
+                    LogFunctionCall("PropertyFunctionsRequiringReflection", objectInstance, args);
+                }
+
                 returnVal = null;
                 return false;
             }
 
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            private void LogFunctionCall(string fileName, object objectInstance, object[] args)
+            {
+                var logFile = Path.Combine(Directory.GetCurrentDirectory(), fileName);
+
+                var argSignature = args?.Aggregate(
+                    string.Empty,
+                    (acc, e) =>$"{acc}{(acc.Equals(string.Empty) ? string.Empty : ", ")}{e?.GetType().Name ?? "null"}");
```

Comment:
Could this be done with

```c#
string.Join(", ", args?.Select(a => a?.GetType().Name ?? "null") ?? Array.Empty<string>);
```

? It tripped my have-to-think-hard-about this meter (even though I love a good map-reduce!). But I'm not sure this is that much better.

Reply:
Yeah, yours looks better. :)